pkg/process/metadata/workloadmeta/extractor_test.go

```diff
@@ -33,31 +32,23 @@ func testProc(pid int32, cmdline []string) *procutil.Process {
 }
 
 func TestExtractor(t *testing.T) {
-	extractor := NewWorkloadMetaExtractor()
-	mockGrpcListener := new(mockGrpcListener)
-	extractor.grpcListener = mockGrpcListener
```

Comment:
I've left more details on the `pkg/process/metadata/workloadmeta/workloadmeta.go` but I've refactored the `WorkloadMetaExctractor` to decouple it from the `gRPCServer`.

By doing so, we don't need to use a mocked gRPC server to test the correct behaviour of the `Extractor`.